channels/set.go

```diff
@@ -167,3 +167,35 @@ func (s Set) String() string {
 	sort.Strings(keys)
 	return fmt.Sprintf("{%s}", strings.Join(keys, ", "))
 }
+
+func (s Set) BuildRedactor(function func(interface{}) base.RedactorFunc) base.Redactor {
+	return RedactorSet{
+		set:          s,
+		redactorFunc: function,
+	}
+}
+
+type RedactorSet struct {
+	set          Set
+	redactorFunc func(interface{}) base.RedactorFunc
+}
+
+func (redactorSet RedactorSet) Redact() string {
+	return redactorSet.GetRedactionString(true)
+}
+
+func (redactorSet RedactorSet) String() string {
+	return redactorSet.GetRedactionString(false)
+}
+
+func (redactorSet RedactorSet) GetRedactionString(shouldRedact bool) string {
+	if !shouldRedact {
+		return redactorSet.set.String()
+	}
+	keys := make([]string, 0, len(redactorSet.set))
+	for channelID := range redactorSet.set {
+		keys = append(keys, fmt.Sprintf("%d.%s", channelID.CollectionID, redactorSet.redactorFunc(channelID.Name).Redact()))
+	}
+	sort.Strings(keys)
+	return fmt.Sprintf("{%s}", strings.Join(keys, ", "))
```

Comment:
If we don't need the sort (and I don't think we do, since we're not using for base.Set), I think it makes sense to optimize this function in the same way we're doing in base.RedactorSet.GetRedactionString:  avoid the use of Sprintf (which has overhead to parse the format) and avoid strings.Join (which is second iteration over the set).

Reply:
The `base.Set` code which I copied this from is self inconsistent - `base.UD(set)` will not sort but `base.Set.String()` will sort the strings! 

I decided, arbitrarily that all channel data is de facto user data so even the standard string printing will show the redaction. This was for optimization.